packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ExtensibleEnumSerializationProvider.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using static Microsoft.Generator.CSharp.Snippets.Snippet;
+
+namespace Microsoft.Generator.CSharp.ClientModel.Providers
+{
+    /// <summary>
+    /// This defines a class with extension methods for enums to convert an enum to its underlying value, or from its underlying value to an instance of the enum
+    /// </summary>
+    internal partial class ExtensibleEnumSerializationProvider : TypeProvider
+    {
+        private readonly InputEnumType _enumType;
+        private TypeProvider? _enumProvider;
+        private TypeProvider EnumProvider => _enumProvider ??= ClientModelPlugin.Instance.TypeFactory.CreateEnum(_enumType);
+
+        protected override string GetNamespace() => ClientModelPlugin.Instance.Configuration.ModelNamespace;
+
+        public ExtensibleEnumSerializationProvider(InputEnumType enumType)
+        {
+            Debug.Assert(enumType.IsExtensible);
+            _enumType = enumType;
+        }
+
+        protected override string BuildRelativeFilePath()
+        {
+            return Path.Combine("src", "Generated", "Models", $"{EnumProvider.Name}.Serialization.cs");
```

Comment:
When `model-namespace` is true, we will put all the model files in subdirectory `Models`, otherwise not.

Does MGC still follow this rule? if so, then we will not add `Models` all time

Reply:
which directory of the source file does not really affect the namespace - we made the namespace properly, and I think maybe always putting them inside a `Models` directory could be an improvement.

Maybe you could open an issue to track if we want to put the models in `Models` directory when their namespace does not contain `Models`.

Reply:
No we always use the model folder regardless of model-namespace.  I think this is a better design for version history in git when you turn on or off model-namespace.  The folder structure does not always directly match the namespace.